lms/validation/_lti_launch_params.py

```diff
@@ -146,32 +143,6 @@ def handle_error(self, error, data, *, many, **kwargs):
         super().handle_error(error, data, many=many, **kwargs)
 
 
-class URLConfiguredBasicLTILaunchSchema(BasicLTILaunchSchema):
-    """Schema for URL-configured basic LTI launches."""
-
-    url = fields.Str(required=True)
-
-    @post_load
-    def _decode_url(self, _data, **_kwargs):  # pylint:disable=no-self-use
-        # Work around a bug in Canvas's handling of LTI Launch URLs in
-        # SpeedGrader launches. In that context, query params get
-        # doubly-encoded. This is worked around by detecting when this has
-        # happened and decoding the URL a second time.
-        #
-        # See https://github.com/instructure/canvas-lms/issues/1486
-        url = _data["url"]
-        if (
-            url.lower().startswith("http%3a")
-            or url.lower().startswith("https%3a")
-            or url.lower().startswith("canvas%3a")
-            or url.lower().startswith("vitalsource%3a")
-        ):
-            url = unquote(url)
-            _data["url"] = url
-
-        return _data
```

Comment:
This is a prime example of non-helpful use of marshmallow. We actually don't use this to check that a URL is there. We already know it is by the time we get here.

This is purely just hiding the logic around URL decoding away from ourselves where it's used.